uber/site_sections/registration.py

```diff
@@ -413,18 +406,21 @@ def check_in_form(self, session, id):
         }
 
     @ajax
-    def check_in(self, session, message='', **params):
+    def check_in(self, session, message='', group_id='', **params):
         attendee = session.attendee(params, allow_invalid=True)
-        group = session.group(attendee.group_id) if attendee.group_id else None
+        group = attendee.group or (session.group(group_id) if group_id else None)
```

Comment:
We still want the attendee's group if it already exists so that we can run validations on it.  Otherwise we just get the group passed in the parameter if it exists.